smart-contracts/assembly/contracts/NFT/__tests__/NFT.spec.ts

```diff
@@ -141,43 +141,70 @@ describe('NFT contract TEST', () => {
     expect(ownerOf(u64ToBytes(tokenToSend))).toStrictEqual(
       stringToBytes(receiver),
     );
+
+    expect(isAllowanceCleared(tokenToSend)).toBeTruthy();
   });
 
   test('approval', () => {
     const tokenId: u64 = 1;
     const addresses = ['2x', '3x'];
 
-    addresses.forEach((address) => {
-      const args = new Args().add(tokenId).add(address).serialize();
-      approve(args);
-    });
-
-    const allowedAddress = bytesToString(
-      getApproved(new Args().add(tokenId).serialize()),
-    );
+    addresses.forEach((address) => approveAddress(tokenId, address));
 
-    const approvedAddressArray = allowedAddress.split(',');
+    const approvedAddressArray = getAllowedAddress(tokenId);
 
     expect(approvedAddressArray[0]).toStrictEqual(addresses[0]);
     expect(approvedAddressArray[1]).toStrictEqual(addresses[1]);
   });
 
   test('transferFrom', () => {
-    const tokenId: u64 = 1;
+    const tokenId: u64 = 3;
     const addresses = ['2x', '3x'];
 
+    addresses.forEach((address) => {
+      approveAddress(tokenId, address);
+    });
+
+    expect(getAllowedAddress(tokenId)).toStrictEqual(addresses);
+
+    const approvedAddressArray = getAllowedAddress(tokenId);
+
+    expect(approvedAddressArray[0]).toStrictEqual(addresses[0]);
+    expect(approvedAddressArray[1]).toStrictEqual(addresses[1]);
+
     transferFrom(
       new Args().add(addresses[0]).add(addresses[1]).add(tokenId).serialize(),
     );
 
-    expect(ownerOf(u64ToBytes(tokenId))).toStrictEqual(
-      stringToBytes(addresses[1]),
-    );
+    expect(isAllowanceCleared(tokenId)).toBeTruthy();
+  });
+
+  test('transferFrom fail if not allowed', () => {
+    const tokenId: u64 = 4;
 
-    const allowedAddress = bytesToString(
-      getApproved(new Args().add(tokenId).serialize()),
+    expect(getAllowedAddress(tokenId)).toStrictEqual([]);
+    expect(ownerOf(u64ToBytes(tokenId))).toStrictEqual(
+      stringToBytes(callerAddress),
     );
 
-    expect(allowedAddress).toStrictEqual('');
+    expect(() => {
+      transferFrom(new Args().add('2x').add('3x').add(tokenId).serialize());
+    }).toThrow();
   });
 });
+
+function isAllowanceCleared(tokenId: u64): boolean {
+  return getAllowedAddress(tokenId).length === 0;
+}
+
+function getAllowedAddress(tokenId: u64): string[] {
+  const allowedAddress = bytesToString(
+    getApproved(new Args().add(tokenId).serialize()),
+  );
+  return allowedAddress === '' ? [] : allowedAddress.split(',');
```

Comment:
now that we can serialize arrays of string, it could be nice to modify getApproved so that is returns an array of addresses instead of a string of addresses separated by coma, WDYT? should we create an issue for this?

Reply:
You are right, it would be better. I'll create an issue.